pkg/apis/intevents/v1beta1/pullsubscription_lifecycle.go

```diff
@@ -85,14 +87,21 @@ func (s *PullSubscriptionStatus) MarkNoSubscription(reason, messageFormat string
 
 // PropagateDeploymentAvailability uses the availability of the provided Deployment to determine if
 // PullSubscriptionConditionDeployed should be marked as true or false.
-func (s *PullSubscriptionStatus) PropagateDeploymentAvailability(d *appsv1.Deployment) {
+// For authentication check purpose, this method will return true  if a false condition
+// is caused by deployment's replicaset unavailable.
+// ReplicaSet unavailable is a sign for potential authentication problems.
+func (s *PullSubscriptionStatus) PropagateDeploymentAvailability(d *appsv1.Deployment) bool {
 	deploymentAvailableFound := false
+	replicaUnavailable := false
 	for _, cond := range d.Status.Conditions {
 		if cond.Type == appsv1.DeploymentAvailable {
 			deploymentAvailableFound = true
 			if cond.Status == corev1.ConditionTrue {
 				pullSubscriptionCondSet.Manage(s).MarkTrue(PullSubscriptionConditionDeployed)
 			} else if cond.Status == corev1.ConditionFalse {
+				if cond.Reason == replicaUnavailableReason {
+					replicaUnavailable = true
+				}
```

Comment:
I changed lifecycle file for v1 pullsubscription, should I change v1beta1 like this?

Reply:
I think we normally try to keep them all in sync, so I think we should change the v1beta1 and v1alpha1 versions as well.

Reply:
pullsubscription does not have v1alpha1 in intevents (v1alpha1 folder only has brokercell). I changed v1 and v1beta1.